provider/global_secondary_index.go

```diff
@@ -328,26 +338,37 @@ func dynamoDBGSIUpdate(d *schema.ResourceData, m interface{}) error {
 		return err
 	}
 
-	if d.HasChange("autoscaling_enabled") && d.Get("autoscaling_enabled").(bool) {
-		if _, err := c.UpdateTable(&dynamodb.UpdateTableInput{
-			TableName: aws.String(tn),
-			GlobalSecondaryIndexUpdates: []*dynamodb.GlobalSecondaryIndexUpdate{
-				&dynamodb.GlobalSecondaryIndexUpdate{
-					Update: &dynamodb.UpdateGlobalSecondaryIndexAction{
-						IndexName: aws.String(in),
-						ProvisionedThroughput: &dynamodb.ProvisionedThroughput{
-							ReadCapacityUnits:  aws.Int64(int64(d.Get("read_capacity").(int))),
-							WriteCapacityUnits: aws.Int64(int64(d.Get("write_capacity").(int))),
-						},
+	if d.HasChange("autoscaling_enabled") && !d.Get("autoscaling_enabled").(bool) {
+		update := &dynamodb.UpdateGlobalSecondaryIndexAction{
+			IndexName:             aws.String(in),
+			ProvisionedThroughput: &dynamodb.ProvisionedThroughput{},
+		}
+
+		changed := false
+		if d.HasChange("read_capaciity") {
+			changed = true
+			update.ProvisionedThroughput.ReadCapacityUnits = aws.Int64(int64(d.Get("read_capacity").(int)))
+		}
+		if d.HasChange("write_capacity") {
+			changed = true
+			update.ProvisionedThroughput.WriteCapacityUnits = aws.Int64(int64(d.Get("write_capacity").(int)))
+		}
+
+		if changed {
+			if _, err := c.UpdateTable(&dynamodb.UpdateTableInput{
+				TableName: aws.String(tn),
+				GlobalSecondaryIndexUpdates: []*dynamodb.GlobalSecondaryIndexUpdate{
+					&dynamodb.GlobalSecondaryIndexUpdate{
+						Update: update,
 					},
 				},
-			},
-		}); err != nil {
-			return err
-		}
+			}); err != nil {
+				return err
+			}
 
-		if err := waitDynamoDBGSIActive(c, tn, in); err != nil {
-			return fmt.Errorf("error waiting for DynamoDB GSI (%s) update on table %s: %w", in, tn, err)
+			if err := waitDynamoDBGSIActive(c, tn, in); err != nil {
```

Comment:
Unrelated to the fixes in this PR. I've seen GSI creation sometimes take several hours or more. We might need an option to not wait for the GSI to become active.

Reply:
That's a good point. Maybe creating state is acceptable. I will look into it next.